app/services/usps_exporter.rb

```diff
@@ -0,0 +1,46 @@
+class UspsExporter
+  OTP_MAX_VALID_DAYS = 30
+
+  def initialize(csv_file_path)
+    @csv_file_path = csv_file_path
+  end
+
+  def run
+    CSV.open(csv_file_path, 'wb', col_sep: '|') do |csv|
+      entries.each do |entry|
+        csv << make_entry_row(entry)
+      end
+    end
+    clear_entries
+  end
+
+  private
+
+  attr_reader :csv_file_path
+
+  def entries
+    UspsConfirmation.all.map(&:decrypted_entry)
+  end
+
+  def clear_entries
+    UspsConfirmation.destroy_all
```

Comment:
I think this is too imprecise, if a conformation is created while this job runs, it will be deleted before it gets enqueued.

I think we should preserve the IDs of the rows we write to the CSV and then delete them, ex

```ruby
entry_ids = entries.map(&:id)
write_to_csv
UspsConfirmation.where(id: entry_ids).delete_all
```